hal-core/src/lib.rs

```diff
@@ -1,5 +1,7 @@
 #![cfg_attr(target_os = "none", no_std)]
+#![feature(try_trait)]
```

Comment:
What do we need this for?

Reply:
I no longer remember. I suspect that means "we don't".

Reply:
ah, it's needed so that we can refer to `core::option::NoneError` here:
https://github.com/hawkw/mycelium/blob/87881769d12019c64e51c69e9b3c8489866f47e4/hal-core/src/mem/page.rs#L372-L376

which...i think we can just remove.